File: lab spring25/assignment4/taskA.cpp
#include <stdio.h>
#include <vector>

#define br printf("\n")
#define rept(i,a,b) for(int i=a;i<b;i++)


void solve(){
    int n,m;
    int matrix[100+1][100+1]={};
    scanf("%d %d",&n, &m);
    int u,v,w;
    while (m--){
        scanf("%d %d %d",&u,&v,&w);
        matrix[u][v] = w;
    }
 
    rept(i,1,n+1) {
        rept(j,1,n+1) {
            printf("%d ",matrix[i][j]);
        }br;
    }
}

int main(){
    freopen("input.txt","r",stdin);
    int t=1;
    // scanf("%d",&t);
    while (t--) solve();
    return 0;
}

File: lab summer25/assignment5/taskA.py
import sys
from collections import deque

sys.stdin = open("input.txt","r")

def bfs(graph,n,start=1):
    q = deque()
    visited = [False]*(n+1)
    q.append(start)
    visited[start] = True

    while q:
        at = q.popleft()
        print(at, end=' ')
        for v in graph[at]:
            if not visited[v]:
                q.append(v)
                visited[v]=True


def solve():
    n,m = list(map(int,input().split()))
    graph = [[] for _ in range(n+1)]
    for _ in range(m):
        u,v = list(map(int,input().split()))
        graph[u].append(v)
        graph[v].append(u)
    bfs(graph,n)

t = 1
t = int(input())
while t>0:
    t-=1
    solve()
    print()